lightbulb/commands/options.py

```diff
@@ -94,14 +98,46 @@ class OptionData(t.Generic[D]):
     """Whether autocomplete is enabled for the option."""
     autocomplete_provider: hikari.UndefinedOr[AutocompleteProviderT] = hikari.UNDEFINED
 
-    localizations: t.Any = hikari.UNDEFINED  # TODO
-    """TODO"""
+    def __post_init__(self) -> None:
+        if len(self.name) < 1 or len(self.name) > 32:
+            raise ValueError("'name' - must be 1-32 characters")
+        if len(self.description) < 1 or len(self.description) > 100:
+            raise ValueError("'description' - must be 1-100 characters")
+
+        if self.choices is not hikari.UNDEFINED:
+            if len(self.choices) > 25:
+                raise ValueError("'choices' - there cannot be more than 25 choices")
+
+            for i, choice in enumerate(self.choices):
+                if len(choice.name) < 1 or len(choice.name) > 100:
+                    raise ValueError(f"'choices[{i}]' - name must be 1-100 characters")
+                if isinstance(choice.value, str) and len(choice.value) > 100:
+                    raise ValueError(f"'choices[{i}]' - value must be <= 100 characters")
+
+        if self.type is hikari.OptionType.STRING:
+            if self.min_length is not hikari.UNDEFINED and (self.min_length < 0 or self.min_length > 6000):
```

Comment:
Can do the same thing here
```suggestion
            if self.min_length is not hikari.UNDEFINED and (0 <= self.min_length <= 6000):
```